Token refresh, settings popup, EPG support

Janhouse opened the pull request:
Sāku nedaudz restrukturēt, lai atcerētos kā tas strādā. Pagaidām pieliku token check pie kanālu listes ģenerēšanas. Reizi dienā tokens tiks force-refreshots, lai atrisinātu #10.
Pēc addona uzinstalēšanas vajadzētu automātiski atvērties settingu logam. Atverot sarakstu vai kanālu tas turpinās parādīties, kamēr lietotājs neievadīs user/pass.

EPG faili tiek pārģenerēti dažas reizes dienā (ar tekošās un nākamās dienas saturu). Settingos ir poga "Configure EPG", kas pārraksta pāri iptvsimple addona settingiem. Pēc izmaiņām jāpārstartē Kodi.

Janhouse committed:
Update .gitignore

Janhouse committed:
Cleanup and formatting

Janhouse committed:
Some refactoring, automatic configuration popup after installing, added login check and periodic token update to fix issue #10

Janhouse committed:
Updated tests and readme.

Janhouse committed:
Update README.md

Janhouse committed:
Add action in settings to clear authorization token.

Janhouse committed:
Added .travis.yml

Janhouse committed:
Pass test credentials through environment.

Janhouse committed:
Allow to skip regionally locked functionality tests

Janhouse committed:
Added initial EPG support through PVR IPTVSimple addon.

Janhouse commented:
Var patestēt vai strādā uz Windows/Linux/Android. Ir pieejams manā Kodi repo: https://github.com/Janhouse/kodi-repo-lattelecomtv
Izskatās, ka arhīva supportam caur TV sadaļu, kā arī citām jaukām fīčām būs jāekstendo iptvsimple addons. Arhīvu gan var viegli pievienot vecajā metodē.

<img width="1278" alt="Screenshot 2019-03-11 at 21 33 05" src="https://user-images.githubusercontent.com/1036439/54152392-ce889300-4445-11e9-93fb-3797cfc79e93.png">
<img width="1282" alt="Screenshot 2019-03-11 at 21 33 53" src="https://user-images.githubusercontent.com/1036439/54152396-cfb9c000-4445-11e9-91e5-197359a69504.png">

#11

reinisb commented:
Tikko notestēju uz macOS 10.14.3. Ieinstalēju tīru Kodi 18.1 un addonu no tava repo. Instalējot lattelecom addonu uzreiz piedāvāja ieinstalēt arī PVR IPTV Simple Client addonu, kā arī atvēra lattelecom addona konfigu. Konfigā ievadīju lattelecom credentials, nospiedu automātisku PVR IPTV Simple Client addona konfigu, pēc tam norestartēju Kodi un lattelecom feed caur Kodi native TV sadaļu strādā lieliski, arī EPG rādās korekti - kopumā ir baigi labi - liels paldies!

Par arhīvu - šo baigi vajadzētu, izrietoši domāju, ka mierīgi var pievienot vecajā metodē.

Par PVR IPTV Simple Client - vai ir zināms vai tas stream atskaņošanai izmanto [inputstream.adaptive](https://github.com/peak3d/inputstream.adaptive)? Konteksts šeit ir tāds, ka kodi by default stream atskaņošanai izmanto iebūvēto ffmpeg, un izrietoši strīms ilgi ielādējas un vēlāk mēdz ķerties, taču ar inputstream.adaptive stream atveras momentāli un pēc tam arī neķeras. Es pēdējo pusgadu šo lattelecom addonu lietoju ar modificētu play_channel() funkciju, kas izmanto inputstream.adaptive - strādā ideāli - varu padalīties ar kodu, ja nepieciešams.

Janhouse committed:
Use inputstream.adaptive to increase performance.

Janhouse committed:
Updated README.

Janhouse committed:
Fix exception upon Kodi startup

Janhouse committed:
Fix for timezone offset (assuming that Lattelecom always provides dates in Europe/Riga timezone).
This could still be buggy during the days when daylight saving time changes happen and it could also require Kodi restart but it looks like there is a problem with Kodi itself as well.

Janhouse committed:
Timezone fix

Janhouse committed:
Added description, category and image to EPG

Janhouse committed:
Add dependency to pytz library

Janhouse committed:
Updated readme and added new screenshots.

Janhouse committed:
Fix tests

Janhouse committed:
Added missing test dependency
